manifests/params.pp

```diff
@@ -92,13 +92,18 @@
   $email_smtp_user_name      = undef
   $email_smtp_password       = undef
 
+  # Defines whether Foreman or the tasks plugin provides the Dynflow executor
+  $dynflow_in_core = true
+
   # OS specific paths
   case $::osfamily {
     'RedHat': {
       $init_config = '/etc/sysconfig/foreman'
       $init_config_tmpl = 'foreman.sysconfig'
       $puppet_etcdir = '/etc/puppet'
       $puppet_home = '/var/lib/puppet'
+      $jobs_service = $dynflow_in_core ? { true => 'dynflow-executor',
```

Comment:
This is tricky. I get what you want to achieve, but currently this is equal to ```$jobs_service = 'dynflow-executor'``` because ```$dynflow_in_core``` is always true. It doesn't look at init.pp.

I'd recommend to drop the ```dynflow_in_core``` parameter altogether and let users override ```$jobs_service```. Some additional documentation in the README can suffice for this. I think this keeps the API and code simpler while allowing users of 1.15 and 1.16 to use it.

The alternative is to default ```jobs_service``` to ```undef``` and determine it in ```foreman::service::jobs``` if undef. Don't forget to change all types to ```Optional[String]```.